src/properties/complete.c

```diff
@@ -0,0 +1,118 @@
+/* -*- mode: C -*-  */
+/* vim:set ts=4 sw=4 sts=4 et: */
+
+/*
+  IGraph library.
+  Copyright (C) 2024 The igraph development team <igraph@igraph.org>
+
+  This program is free software; you can redistribute it and/or modify it under
+  the terms of the GNU General Public License as published by the Free Software
+  Foundation; either version 2 of the License, or (at your option) any later
+  version.
+
+  This program is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along with
+  this program. If not, see <https://www.gnu.org/licenses/>.
+
+*/
+
+
+#include "igraph_interface.h"
+#include "igraph_structural.h"
+
+#include "graph/internal.h"
+
+/**
+ * \ingroup structural
+ * \function igraph_is_complete
+ * \brief Decides whether the graph is complete.
+ *
+ * A graph is considered complete if all pairs of different vertices are
+ * adjacent.
+ *
+ * This function is currently only implemented on undirected graphs.
+ *
+ * The null graph and the singleton graph are considered complete.
+ *
+ * \param graph The graph object to analyse.
+ * \param res Pointer to a logical variable, the result will be stored here.
+ *
+ * Time complexity: O(1) for simple and sparse graphs, O(|E|) at worst for graphs
+ * with either loops or multiple edges.
+ *
+ * Memory complexity: O(1) for simple and sparse graphs, O(|V|) at worst for
+ * graphs with either loops or multiple edges.
```

Comment:
The time complexity would be O(|V| + |E|), i.e. no better than `igraph_is_simple()`, which this function calls. Of course if it was already cached that the graph is simple, then this is improved. But before knowing that, we can't guarantee anything better.